test/e2e/storage/external/external.go

```diff
@@ -304,17 +317,56 @@ func (d *driverDefinition) GetDynamicProvisionStorageClass(config *testsuites.Pe
 	return testsuites.GetStorageClass(sc.Provisioner, sc.Parameters, sc.VolumeBindingMode, f.Namespace.Name, "e2e-sc")
 }
 
+func loadSnapshotClass(filename string) (*unstructured.Unstructured, error) {
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	snapshotClass := &unstructured.Unstructured{}
+
+	if err := runtime.DecodeInto(scheme.Codecs.UniversalDecoder(), data, snapshotClass); err != nil {
+		return nil, errors.Wrap(err, filename)
+	}
+
+	return snapshotClass, nil
+}
+
 func (d *driverDefinition) GetSnapshotClass(config *testsuites.PerTestConfig) *unstructured.Unstructured {
-	if !d.SnapshotClass.FromName {
+	if !d.SnapshotClass.FromName && d.SnapshotClass.FromFile == "" && d.SnapshotClass.FromExistingClassName == "" {
 		e2eskipper.Skipf("Driver %q does not support snapshotting - skipping", d.DriverInfo.Name)
 	}
 
-	snapshotter := d.DriverInfo.Name
-	parameters := map[string]string{}
-	ns := config.Framework.Namespace.Name
-	suffix := snapshotter + "-vsc"
+	var snapshotClass *unstructured.Unstructured
+	var err error
+	f := config.Framework
+
+	switch {
+	case d.SnapshotClass.FromName:
+		snapshotter := d.DriverInfo.Name
+		parameters := map[string]string{}
+		ns := config.Framework.Namespace.Name
+		suffix := snapshotter + "-vsc"
```

Comment:
suffix should just be "-vsc"?

Reply:
Actually, it came from the original code. And it could be both, because it adds `ns` to suffix inside `GetSnapshotClass` to be unique. (If there will be a test case that creates multiple SnapshotClass, it's safe to keep `snapshotter`, though.)

https://github.com/kubernetes/kubernetes/blob/0a2a69add2c914675898bd9018010a03eca8f693/test/e2e/storage/testsuites/driveroperations.go#L100